src/Compilers/Core/Portable/PEWriter/TypeNameSerializer.cs

```diff
@@ -179,7 +179,7 @@ private static void AppendAssemblyQualifierIfNecessary(StringBuilder sb, ITypeRe
             IManagedPointerTypeReference reference = typeReference as IManagedPointerTypeReference;
             if (reference != null)
             {
-                AppendAssemblyQualifierIfNecessary(sb, pointer.GetTargetType(context), out isAssemQualified, context);
+                AppendAssemblyQualifierIfNecessary(sb, reference.GetTargetType(context), out isAssemQualified, context);
```

Comment:
When we get here `pointer` is `null`, because the `if` statement on line 173 returns if `pointer` is not `null`.